Automattic/pocket-casts-android
# Migrate AutoArchiveFragment to Jetpack Compose

## Description
- This PR migrate the global AutoArchive screen from xml to Jetpack Compose
- The Dialog with this update looks a bit different since I am updating to use the same JC dialog component that we are using in the other's screen

## Testing Instructions
1. Go to Profile -> Settings -> Auto Archive
2. ✅ Ensure `settings_auto_archive_shown` is tracked when opening Auto Archive screen
3. Change Archive playing episodes
4. ✅ Ensure `settings_auto_archive_played_changed, Properties: {"value":"[value]"` is tracked
5. Change Archive inactive episodes
6. ✅ Ensure `settings_auto_archive_inactive_changed, Properties: {"value":"[value]"` is tracked
7. Toggle Include starred episodes
8. ✅ Ensure `settings_auto_archive_include_starred_toggled, Properties: {"enabled":[value]` is tracked
9. Leave the screen
10. Back to the screen again
11. ✅ Ensure all values you set still persist
12. Test all themes
13. Smoke test the feature


## Screenshots or Screencast 
| Before | Before |
|--------|--------|
| ![Screenshot_20240604_085745](https://github.com/Automattic/pocket-casts-android/assets/42220351/4f9aafb4-1824-4785-ac6d-6b6d53914a47) | ![Screenshot_20240604_085752](https://github.com/Automattic/pocket-casts-android/assets/42220351/9830c79e-34cb-4bc1-9d91-fecb9048b33c) | 


| After | After |
|--------|--------|
| ![new](https://github.com/Automattic/pocket-casts-android/assets/42220351/82a7a575-2522-4d1e-8d15-76d7abbd623e) | ![new2](https://github.com/Automattic/pocket-casts-android/assets/42220351/2621cff9-225c-4bac-a156-bf54307ccb8e) | 






## Checklist
 
#### I have tested any UI changes...
